src/parameterizations/lateral/MOM_lateral_mixing_coeffs.F90

```diff
@@ -1128,6 +1153,123 @@ subroutine calc_slope_functions_using_just_e(h, G, GV, US, CS, e)
 
 end subroutine calc_slope_functions_using_just_e
 
+!! Computes UH_grad and VH_grad for gradient model (Khani & Dawson, JAMES 2023)
+subroutine calc_slope_functions_with_gradient_model(h, G, GV, US, CS, e, uh, vh)
+  type(ocean_grid_type),                       intent(inout) :: G  !< Ocean grid structure
+  type(verticalGrid_type),                     intent(in)    :: GV !< Vertical grid structure
+  real, dimension(SZI_(G),SZJ_(G),SZK_(GV)),   intent(inout) :: h  !< Layer thickness [H ~> m or kg m-2]
+  real, dimension(SZIB_(G),SZJ_(G),SZK_(GV)),  intent(in) :: uh    !< Interface height times u [ZU ~> m2 s-1]
```

Comment:
The `uh` and `vh` variables that are being passed in to calls to this routines actually are thickness transports through faces and have units of `[H L2 T-1 ~> m3 s-1 or kg s-1]`.